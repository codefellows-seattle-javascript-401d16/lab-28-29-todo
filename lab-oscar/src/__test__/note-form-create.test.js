import React from 'react';
import {shallow} from 'enzyme';
import NoteCreateForm from '../component/note-create-form';

describe('testing noteCreateForm', () => {
  test('it should have correct default state values', () => {
    let wrapper = shallow(<NoteCreateForm handleNoteCreate={() => {}} />);
    expect(wrapper.state('content')).toBe('');
  });
  test('expect submit to invoke handleNoteCreate', () => {
    let noteCreate = (note) => {
      expect(note.title).toEqual('');
    };
    let wrapper = shallow(<NoteCreateForm handleNoteCreate={noteCreate} />);
    wrapper.find('form').simulate('click');
  });
  test('testing onChange handler', () => {
    const component = shallow(<NoteCreateForm handleNoteCreate={() => {}} />);
    component.find('input').simulate('change', {target: { value: 'abc'}});
    expect(component).toMatchSnapshot();
  });
});
