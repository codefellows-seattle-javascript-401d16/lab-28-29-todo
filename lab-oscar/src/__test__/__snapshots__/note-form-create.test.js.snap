// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing noteCreateForm testing onChange handler 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="input-form"
>
    <form
        onSubmit={[Function]}
    >
        <input
            name="content"
            onChange={[Function]}
            placeholder="What needs to done?"
            type="text"
            value=""
        />
        <button
            type="submit"
        >
            Add Note 
        </button>
    </form>
</div>,
  "nodes": Array [
    <div
      className="input-form"
>
      <form
            onSubmit={[Function]}
      >
            <input
                  name="content"
                  onChange={[Function]}
                  placeholder="What needs to done?"
                  type="text"
                  value=""
            />
            <button
                  type="submit"
            >
                  Add Note 
            </button>
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NoteCreateForm
        handleNoteCreate={[Function]}
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NoteCreateForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "handleSubmit": [Function],
        "props": Object {
          "handleNoteCreate": [Function],
        },
        "refs": Object {},
        "state": Object {
          "content": "",
          "undefined": "abc",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="input-form"
>
          <form
                    onSubmit={[Function]}
          >
                    <input
                              name="content"
                              onChange={[Function]}
                              placeholder="What needs to done?"
                              type="text"
                              value=""
                    />
                    <button
                              type="submit"
                    >
                              Add Note 
                    </button>
          </form>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="input-form"
>
          <form
                    onSubmit={[Function]}
          >
                    <input
                              name="content"
                              onChange={[Function]}
                              placeholder="What needs to done?"
                              type="text"
                              value=""
                    />
                    <button
                              type="submit"
                    >
                              Add Note 
                    </button>
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NoteCreateForm
    handleNoteCreate={[Function]}
/>,
}
`;
